engine/core/src/main/java/org/datacleaner/configuration/DataCleanerConfigurationImpl.java

```diff
@@ -0,0 +1,143 @@
+            DatastoreCatalog datastoreCatalog, ReferenceDataCatalog referenceDataCatalog) {
+        if (environment == null) {
+            _environment = defaultEnvironment();
+        } else {
+            _environment = environment;
+        }
+        if (homeFolder == null) {
+            _homeFolder = defaultHomeFolder();
+        } else {
+            _homeFolder = homeFolder;
+        }
+        if (datastoreCatalog == null) {
+            _datastoreCatalog = defaultDatastoreCatalog();
+        } else {
+            _datastoreCatalog = datastoreCatalog;
+        }
+        if (referenceDataCatalog == null) {
+            _referenceDataCatalog = defaultReferenceDataCatalog();
+        } else {
+            _referenceDataCatalog = referenceDataCatalog;
+        }
+    }
+
+    public DataCleanerConfigurationImpl withDatastoreCatalog(DatastoreCatalog datastoreCatalog) {
+        return new DataCleanerConfigurationImpl(getEnvironment(), getHomeFolder(), datastoreCatalog,
+                getReferenceDataCatalog());
+    }
+
+    public DataCleanerConfigurationImpl withDatastores(Datastore... datastores) {
+        return withDatastoreCatalog(new DatastoreCatalogImpl(datastores));
+    }
+
+    public DataCleanerConfigurationImpl withDatastores(Collection<Datastore> datastores) {
+        return withDatastoreCatalog(new DatastoreCatalogImpl(datastores));
+    }
+
+    public DataCleanerConfigurationImpl withReferenceDataCatalog(ReferenceDataCatalog referenceDataCatalog) {
+        return new DataCleanerConfigurationImpl(getEnvironment(), getHomeFolder(), getDatastoreCatalog(),
+                referenceDataCatalog);
+    }
+
+    public DataCleanerConfigurationImpl withHomeFolder(RepositoryFolder homeFolder) {
+        return new DataCleanerConfigurationImpl(getEnvironment(), homeFolder, getDatastoreCatalog(),
+                getReferenceDataCatalog());
+    }
+
+    public DataCleanerConfigurationImpl withEnvironment(DataCleanerEnvironment environment) {
+        return new DataCleanerConfigurationImpl(environment, getHomeFolder(), getDatastoreCatalog(),
+                getReferenceDataCatalog());
+    }
+    
+    @Override
+    public DatastoreCatalog getDatastoreCatalog() {
+        return _datastoreCatalog;
+    }
+
+    @Override
+    public ReferenceDataCatalog getReferenceDataCatalog() {
+        return _referenceDataCatalog;
+    }
+
+    @Override
+    public RepositoryFolder getHomeFolder() {
+        return _homeFolder;
+    }
+
+    @Override
+    public DataCleanerEnvironment getEnvironment() {
+        return _environment;
+    }
+
+    public static ReferenceDataCatalog defaultReferenceDataCatalog() {
+        return new ReferenceDataCatalogImpl();
+    }
+
+    public static DatastoreCatalog defaultDatastoreCatalog() {
+        return new DatastoreCatalogImpl();
+    }
+
+    public static RepositoryFolder defaultHomeFolder() {
+        return new FileRepository(".");
```

Comment:
It is still the default if you just instantiate a new DataCleanerConfigurationImpl. This is still fine behaviour for casual use of the class (say, in a unittest or so) but in the two "real" environments ("desktop" and "monitor") we set another home folder that fits those situations.